### ISSUE TITLE
Make Drag Threshold configurable

### ISSUE THREAD

**Issue by Tscheffrey** [davidjerleke/embla-carousel#25]
As already mentioned in #22, I think it would be desirable to make the threshold at which the slider starts to react to the hand/cursor movement configurable. 

That might look like this:

```javascript
const embla = EmblaCarousel(emblaNode, {
  dragThreshold: 5,
  ...
})
```
With the value being the amount of pixels the hand/cursor has to move until the silder will follow its movement.

Apparently there already is a drag threshold already built in and I'm sure that the default value will be enough for most devs but depending on what kind of device the slider is used on, it could be quite useful (I'm thinking mostly about touch screens here).

Jeff

**Comment by davidcetinkaya** [comment]
Hello Jeff (@Tscheffrey),

Thank you very much for taking time to create this issue 👍🏻.

I’ve tried to gather user opinions on this for a period of time to no avail. I’m going to close this issue for now due to lack of interest.

Kindly,
David

**Comment by davidcetinkaya** [comment]
.

**Comment by sh-a-v** [comment]
@davidcetinkaya this is a very necessary option

**Comment by davidcetinkaya** [comment]
Hello Alexander (@sh-a-v),
Would you care explaining why? What are the benefits of this?

**Comment by silllli** [comment]
I have a use case: I built a slider that features slides which are about 100% of the browser width. They contain images which on the other hand don’t use all of their slide’s horizontal space. I want the mouse interaction only to happen and be visible (by changing the cursor), when the mouse is placed on an image—not on the whitespace around it. It works okay with images in landscape format, but with ones in portrait you have to drag the image unnaturally far to trigger the slide change. Multiple people complained about this, so I would be happy about that option!

**Comment by davidjerleke** [comment]
Hi @silllli,

Thanks for your input. Would you mind creating a CodeSandbox to demonstrate what you mean? I'm not sure I'm following.

**Comment by silllli** [comment]
Yes, something like this: https://codesandbox.io/s/embla-carousel-align-vanilla-forked-7opur4

If you open it on a 27“ screen, you need to drag the slide really far and/or fast. Actually, the effect is also there with images in landscape format, but it’s more obvious with portraits, since the amount of needed dragging is bigger compared to the absolute image width in the browser. So maybe the problem is actually the calculation of the threshold—shouldn’t it feel the same on a 16“ and a 27“ screen?

**Comment by davidjerleke** [comment]
Thanks @silllli,

I see the problem now. With the current implementation, the drag threshold is set to be a specific percentage of the viewport. This is working well on devices from mobile phones ---> small/medium laptops. But as you describe, as screens get bigger it's causing problems.

**My suggestion** would be to keep the current behavior:

- The drag threshold is set to be a specific percentage of the viewport.
- **BUT**, I think we should limit it within a minimum/maximum range, so it doesn't grow out of hand for larger screens than laptops.

I've created a [CodeSandbox](https://codesandbox.io/s/embla-carousel-react-v7-4r5fu9?file=/src/js/embla-carousel-v7.js:6843-6913) where you can try the approach I describe above. Feel free to try it out. Try it on small and large screens and experiment with the `minimumDragThreshold` and `maximumDragThreshold` values until you think it's behaving as expected.

If my suggestion still doesn't solve the problem, I'm willing to discuss other solutions. Thanks for your time!

Best,
David

**Comment by silllli** [comment]
Thank you @davidjerleke for looking at this so promptly. And yes, it makes total sense up to a certain point that the threshold is based on the browser width. I’ll be back at my desk with a large screen tomorrow where I can test it. On the other hand, I bet my judgement wouldn’t be better than yours, so go ahead if you want to implement it before!

**Comment by davidjerleke** [comment]
No rush @silllli, I'll wait for you to test it. Cheers!

**Comment by silllli** [comment]
Sorry that it took that long, I finally got to it:

I think it’s a bit tricky since it depends on the input device. With an external mouse, a higher value (like 250) seems to work, but that’s still hard to handle on a trackpad. On a trackpad I would even go as low as 150. What do you think?

(In you CodeSandbox, somehow the threshold seems to be a lot higher between the first and last slide than in between?)

**Comment by davidjerleke** [comment]
Thanks a lot for looking into this @silllli.

> I think it’s a bit tricky since it depends on the input device. With an external mouse, a higher value (like 250) seems to work, but that’s still hard to handle on a trackpad. On a trackpad I would even go as low as 150. What do you think?

Yes that seems tricky. I found this [StackOverflow](https://stackoverflow.com/a/62415754/4647781) solution to detect if the input device is a mouse or a trackpad, but if you read the comments the solution doesn't seem to be waterproof. It's also har to test extensively because you would need access to a lot of devices and also try these devices in different browsers 😕. 

I don't know where to go from here. At least we can agree that the drag threshold should be limited to a max value so it's not impossible to change slides on large screens. What's your thoughts?

I'm also thinking that making drag threshold configurable wouldn't solve this problem you're describing.

> (In you CodeSandbox, somehow the threshold seems to be a lot higher between the first and last slide than in between?)

Hmm. Not sure. I'll have to try this and see if something strange is going on.

Best,
David

**Comment by silllli** [comment]
> I don't know where to go from here. At least we can agree that the drag threshold should be limited to a max value so it's not impossible to change slides on large screens. What's your thoughts?

I guess most important is to have a limit at all. Also, we could assume that most people that use a 27" screen (or larger) also use an external mouse, so there is no need to worry that much about this combination. (Well… I use my laptop’s trackpad and an external mouse interchangeably, together with a 27" screen.)

Maybe simply set the limit to something like 225?

> I'm also thinking that making drag threshold configurable wouldn't solve this problem you're describing.

No, I don’t that’s needed.

Best,
Silvan

**Comment by davidjerleke** [comment]
Thanks for your help @silllli 👍! I'll have this on my roadmap for the next release.

**Comment by wolfgangschoeffel** [comment]
@davidjerleke Thank’s for the quick reply. 

If clicking elements in the carousel is not necessary, the threshold could also stay low. So it could also be a boolean if you don’t want to expose the threshold value directly but in the other libraries I found it quite handy to be able to dial it in depending on the situation.

On touch (as far as I can tell) it works very nicely.

**Comment by davidjerleke** [comment]
Hi @wolfgangschoeffel,

Thanks for your input. I’ve decided to expose a `dragThreshold` option but can say when I have time to look into this. Note that this will only affect when clicks are fired and not. In contrast to other carousel libraries, it will **not** affect when dragging of the carousel starts.

Until I’ve implemented this, I encourage you to check out the new [v8.0.0-rc01 release](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc01) and all its new features.

Best,
David

**Comment by davidjerleke** [comment]
@wolfgangschoeffel and anyone interested in this feature. It will be released with v8.0.0-rc03.

**Comment by wolfgangschoeffel** [comment]
Amazing. Thank you @davidjerleke! Looking forward to try it out.

**Comment by davidjerleke** [comment]
@wolfgangschoeffel I just released [v8.0.0-rc03](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc03).
